modules/apps/wiki/wiki-service/src/main/java/com/liferay/wiki/service/impl/WikiNodeLocalServiceImpl.java

```diff
@@ -113,9 +123,16 @@ public WikiNode addNode(
 
 		long nodeId = counterLocalService.increment();
 
+		if (externalReferenceCode == null) {
+			externalReferenceCode = String.valueOf(nodeId);
+		}
+
+		_validateExternalReferenceCode(externalReferenceCode, groupId);
+
 		WikiNode node = wikiNodePersistence.create(nodeId);
 
 		node.setUuid(serviceContext.getUuid());
+		node.setExternalReferenceCode(externalReferenceCode);
 		node.setGroupId(groupId);
```

Comment:
uhhhh things that are not sorted... let's leave them this way.

Reply:
In this case, this is the order that makes sense for Brian because is the same order as the Service Builder DTD (he modified it to make it like this in the MBMessage PR)